From 4e66114109263fff6b13192e07be9bbd9e493fee Mon Sep 17 00:00:00 2001
From: Mark Reynolds <mreynolds@redhat.com>
Date: Mon, 17 Apr 2017 17:06:19 -0400
Subject: [PATCH 1/2] Issue 49221 - During an upgrade the provided localhost
 name is ignored

Description:  If the FullMachine name, or localhost, is provided in an INF
              it is ignored during the upgrade the value of nsslapd-localhost
              from the current server is used instead.  We should only override
              the localhost value if it is missing.

https://pagure.io/389-ds-base/issue/49221

Reviewed by: nhosoi(Thanks!)
---
 ldap/admin/src/scripts/DSUpdate.pm.in | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ldap/admin/src/scripts/DSUpdate.pm.in b/ldap/admin/src/scripts/DSUpdate.pm.in
index e84a9a9..8b24b47 100644
--- a/ldap/admin/src/scripts/DSUpdate.pm.in
+++ b/ldap/admin/src/scripts/DSUpdate.pm.in
@@ -435,7 +435,9 @@ sub initInfFromInst {
     my $servid = $inst;
     $servid =~ s/slapd-//;
 
-    $inf->{General}->{FullMachineName} = $entry->getValue("nsslapd-localhost");
+    if (!$inf->{General}->{FullMachineName}) {
+        $inf->{General}->{FullMachineName} = $entry->getValue("nsslapd-localhost");
+    }
     $inf->{General}->{SuiteSpotUserID} = $entry->getValue("nsslapd-localuser");
     $inf->{slapd}->{ServerPort} = $entry->getValue("nsslapd-port");
     $inf->{slapd}->{ldapifilepath} = $entry->getValue("nsslapd-ldapifilepath");
-- 
2.9.3

